pkg/envoy/rds/route/route_config.go

```diff
@@ -192,8 +192,10 @@ func buildInboundRoutes(rules []*trafficpolicy.Rule) []*xds_route.Route {
 func buildOutboundRoutes(outRoutes []*trafficpolicy.RouteWeightedClusters) []*xds_route.Route {
 	var routes []*xds_route.Route
 	for _, outRoute := range outRoutes {
-		emptyHeaders := map[string]string{}
-		routes = append(routes, buildRoute(trafficpolicy.PathMatchRegex, constants.RegexMatchAll, constants.WildcardHTTPMethod, emptyHeaders, outRoute.WeightedClusters, outRoute.RetryPolicy))
+		outRoute.HTTPRouteMatch.PathMatchType = trafficpolicy.PathMatchRegex
+		outRoute.HTTPRouteMatch.Path = constants.RegexMatchAll
+		outRoute.HTTPRouteMatch.Headers = map[string]string{}
```

Comment:
Please create a tmp var instead of modifying the loop variable, which is a pointer type. Doing so can result in surprising bugs such as the one fixed by commit 728205537d5a17264be943a45bfe9fe3a57ad5f6.
It would also make the code easier to comprehend.